enterprise/coderd/workspaceproxycoordinate.go

```diff
@@ -7,42 +7,11 @@ import (
 	"nhooyr.io/websocket"
 
 	"github.com/coder/coder/v2/coderd/httpapi"
-	"github.com/coder/coder/v2/coderd/httpmw"
 	"github.com/coder/coder/v2/coderd/util/apiversion"
 	"github.com/coder/coder/v2/codersdk"
-	"github.com/coder/coder/v2/enterprise/wsproxy/wsproxysdk"
 	"github.com/coder/coder/v2/tailnet/proto"
 )
 
-// @Summary Agent is legacy
-// @ID agent-is-legacy
-// @Security CoderSessionToken
-// @Produce json
-// @Tags Enterprise
-// @Param workspaceagent path string true "Workspace Agent ID" format(uuid)
-// @Success 200 {object} wsproxysdk.AgentIsLegacyResponse
-// @Router /workspaceagents/{workspaceagent}/legacy [get]
-// @x-apidocgen {"skip": true}
-func (api *API) agentIsLegacy(rw http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-
-	agentID, ok := httpmw.ParseUUIDParam(rw, r, "workspaceagent")
-	if !ok {
-		httpapi.Write(r.Context(), rw, http.StatusBadRequest, codersdk.Response{
-			Message: "Missing UUID in URL.",
-		})
-		return
-	}
-
-	node := (*api.AGPL.TailnetCoordinator.Load()).Node(agentID)
-	httpapi.Write(ctx, rw, http.StatusOK, wsproxysdk.AgentIsLegacyResponse{
-		Found: node != nil,
-		Legacy: node != nil &&
-			len(node.Addresses) > 0 &&
-			node.Addresses[0].Addr() == codersdk.WorkspaceAgentIP,
-	})
-}
```

Comment:
We might want to keep this endpoint ahd just always have it return false, otherwise workspace proxies will break during the upgrade process.

Reply:
Workspace proxies treat an error checking if the workspace is legacy as assuming it is. So, they should get a 404, assume it is legacy.

That's not the most efficient, I agree, but they should upgrade their workspace proxies.

I'd prefer to minimize the amount of legacy stuff we keep around.

Reply:
Ah, that seems reasonable.